Pro dummy app broken after pnpm workspace migration: webpack not found and duplicate react-refresh loader

## Problem

PR #2338 (Remove yalc, use pnpm workspaces for dummy apps) left two issues in the Pro dummy app that break local development:

### 1. `Command "webpack" not found` when running Procfile.dev

The Pro dummy app still has its own `pnpm-workspace.yaml` (with just `packages: ["."]`) from the yalc era. When any `pnpm exec` command runs from the dummy directory, pnpm sees this file and treats the dummy as an isolated workspace root — completely cut off from the repo root's `node_modules/.bin/` where webpack is hoisted.

**Error:**
```
ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command "webpack" not found
```

### 2. Duplicate `$ReactRefreshModuleRuntime$` declaration in dev server mode

The dummy's `devDependencies` pins `@pmmmwh/react-refresh-webpack-plugin@0.5.3`, but Shakapacker 9.5.0 (at root) depends on `0.5.17`. pnpm can't hoist both to root, so both copies coexist. In webpack-dev-server mode, the react-refresh loader gets applied twice (each copy has a different absolute path, so the plugin's double-injection guard can't detect the duplicate).

**Error:**
```
Module parse failed: Identifier '$ReactRefreshModuleRuntime$' has already been declared
```

## Fix

1. Delete `react_on_rails_pro/spec/dummy/pnpm-workspace.yaml` — pnpm traverses up to the root workspace and correctly resolves binaries
2. Remove the pinned `@pmmmwh/react-refresh-webpack-plugin@0.5.3` from dummy's `devDependencies` — Shakapacker already provides this package via root

## Related

Follow-up cleanup from #2338 / #2089

--- a/react_on_rails_pro/spec/dummy/package.json
+++ b/react_on_rails_pro/spec/dummy/package.json
@@ -76,7 +76,6 @@
   "devDependencies": {
     "@babel/preset-typescript": "^7.23.2",
     "@playwright/test": "^1.56.1",
-    "@pmmmwh/react-refresh-webpack-plugin": "0.5.3",
     "@tailwindcss/aspect-ratio": "^0.4.2",
     "@tailwindcss/forms": "^0.5.3",
     "@tailwindcss/typography": "^0.5.9",

--- a/react_on_rails_pro/spec/dummy/pnpm-workspace.yaml
+++ b/react_on_rails_pro/spec/dummy/pnpm-workspace.yaml
@@ -1,2 +1,0 @@
-packages:
-  - .

